packages/@aws-cdk/aws-eks/lib/private/bottlerocket.ts

```diff
@@ -0,0 +1,44 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ssm from '@aws-cdk/aws-ssm';
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Properties for BottleRocketImage
+ */
+export interface BottleRocketImageProps {
+  /**
+   * The Kubernetes version to use
+   */
+  readonly kubernetesVersion: string;
+}
+
+/**
+ * Construct an Bottlerocket image from the latest AMI published in SSM
+ */
+export class BottleRocketImage implements ec2.IMachineImage {
+  private readonly kubernetesVersion: string;
+
+  private readonly amiParameterName: string;
+
+  /**
+   * Constructs a new instance of the BottleRocketImage class.
+   */
+  public constructor(props: BottleRocketImageProps) {
+    this.kubernetesVersion = props.kubernetesVersion;
+
+    // set the SSM parameter name
+    this.amiParameterName = `/aws/service/bottlerocket/aws-k8s-${this.kubernetesVersion}/x86_64/latest/image_id`;
+  }
+
+  /**
+   * Return the correct image
+   */
+  public getImage(scope: Construct): ec2.MachineImageConfig {
+    const ami = ssm.StringParameter.valueForStringParameter(scope, this.amiParameterName);
+    return {
+      imageId: ami,
+      osType: ec2.OperatingSystemType.LINUX,
+      userData: ec2.UserData.custom(''),
```

Comment:
Why is this required?

Reply:
> Why is this required?

We had exactly the same discussion here
https://github.com/aws/aws-cdk/pull/7282#discussion_r407020235

Reply:
Can we chance it to optional? (let's at least open an issue?)